lib/std/os/linux.zig

```diff
@@ -6146,7 +6146,256 @@ pub const termios = switch (native_arch) {
     },
 };
 
+pub const SOCK_IOC_TYPE	= 0x89;
+
+pub const SIOCGSTAMP_NEW = IOCTL.IOR(SOCK_IOC_TYPE, 0x06, i64[2]);
+pub const SIOCGSTAMP_OLD = IOCTL.IOR('s', 100, timeval);
+
+/// Get stamp (timeval)
+pub const SIOCGSTAMP = if (native_arch == .x86_64 or @sizeOf(timeval) == 8) SIOCGSTAMP_OLD else SIOCGSTAMP_NEW;
+
+pub const SIOCGSTAMPNS_NEW = IOCTL.IOR(SOCK_IOC_TYPE, 0x07, i64[2]);
+pub const SIOCGSTAMPNS_OLD = IOCTL.IOR('s', 101, kernel_timespec);
+
+/// Get stamp (timespec)
+pub const SIOCGSTAMPNS = if (native_arch == .x86_64 or @sizeOf(timespec) == 8) SIOCGSTAMPNS_OLD else SIOCGSTAMPNS_NEW;
```

Comment:
I wanted to highlight this because I'm not confident whether it's correct.

Especially the if statements so I'm explaining it below:
In https://github.com/torvalds/linux/blob/master/arch/x86/include/uapi/asm/bitsperlong.h
```h
#if defined(__x86_64__) && !defined(__ILP32__)
# define __BITS_PER_LONG 64
#else
# define __BITS_PER_LONG 32
#endif
```
And in https://github.com/torvalds/linux/blob/master/include/uapi/linux/sockios.h
```h
#if __BITS_PER_LONG == 64 || (defined(__x86_64__) && defined(__ILP32__))
/* on 64-bit and x32, avoid the ?: operator */
#define SIOCGSTAMP	SIOCGSTAMP_OLD
#define SIOCGSTAMPNS	SIOCGSTAMPNS_OLD
#else
#define SIOCGSTAMP	((sizeof(struct timeval))  == 8 ? \
			 SIOCGSTAMP_OLD   : SIOCGSTAMP_NEW)
#define SIOCGSTAMPNS	((sizeof(struct timespec)) == 8 ? \
			 SIOCGSTAMPNS_OLD : SIOCGSTAMPNS_NEW)
#endif
```

So it could be simplified to:
```h
#if defined(__x86_64__) || (sizeof(struct timeval))  == 8
#define SIOCGSTAMP SIOCGSTAMP_OLD
#else
#define SIOCGSTAMP SIOCGSTAMP_NEW
#endif

#if defined(__x86_64__) || (sizeof(struct timeval))  == 8
#define SIOCGSTAMPNS SIOCGSTAMPNS_OLD
#else
#define SIOCGSTAMPNS SIOCGSTAMPNS_NEW
#endif
```

Reply:
I believe your simplification is correct.

Reply:
could please also confirm whether I'm using right structs in this section of code: `timeval`, `kernel_timespec` and `timespec`. (if data types match the c version)

Reply:
Yes, that's fine. `__kernel_timespec` and `timespec` really are distinct types.